table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1559,4 +1640,214 @@ protected static String getMillisecondPrecisionZonedDateTime(ZonedDateTime dateT
                 .withZone(ZoneId.of("UTC"))
                 .format(dateTime);
     }
-}
\ No newline at end of file
+
+    @Override
+    public void registerTableListener(String registrationId, Instant newExpirationTime) {
+        Instant now = _clock.instant();
+
+        Delta registrationDelta = TableEventDatacenter.newRegistrant(registrationId, now, newExpirationTime);
+
+        Audit audit = new AuditBuilder()
+                .setComment(String.format("Registering %s", registrationId))
+                .setProgram(registrationId)
+                .setLocalHost()
+                .build();
+
+        _backingStore.update(_systemTableEventRegistry, _selfDataCenter, TimeUUIDs.newUUID(), registrationDelta, audit, WriteConsistency.GLOBAL);
+    }
+
+    @Override
+    public Map.Entry<String, TableEvent> getNextTableEvent(String registrationId) {
+        TableEventDatacenter datacenter = _objectMapper.convertValue(
+                        _backingStore.get(_systemTableEventRegistry, _selfDataCenter, ReadConsistency.STRONG),
+                        TableEventDatacenter.class
+                );
+        java.util.Optional<TableEventRegistrant> registrant = java.util.Optional.ofNullable(datacenter.getRegistrants().get(registrationId));
+
+        return registrant
+                .map(TableEventRegistrant::getTasks)
+                .orElse(Collections.emptyMap())
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getValue().isReady())
+                .findAny()
+                .orElse(null);
+    }
+
+    @Override
+    public Stream<String> getIdsForStorage(String table, String uuid) {
+
+        checkNotNull(_storageReaderDAO);
+
+        Storage uuidStorage = readTableJson(table, false).getStorages().stream()
+                .filter(storage -> storage.getUuidString().equals(uuid))
+                .findFirst()
+                .get();
+
+        return _storageReaderDAO.getIdsForStorage(newAstyanaxStorage(uuidStorage, table));
```

Comment:
I like the way how you have the getIdsForStorage in two different interfaces and implementing them as per your requirement.